apps/api/src/app/workflows-v2/usecases/list-workflows/list-workflow.usecase.ts

```diff
@@ -0,0 +1,40 @@
+import { BadRequestException, Injectable } from '@nestjs/common';
+
+import { EnvironmentRepository, NotificationTemplateRepository } from '@novu/dal';
+import { ListWorkflowsCommand } from './list-workflows.command';
+import { ListWorkflowResponse } from '../../dto/workflow-commons-fields';
+import { NotificationTemplateMapper } from '../../mappers/notification-template-mapper';
+
+@Injectable()
+export class ListWorkflowsUseCase {
+  constructor(
+    private notificationTemplateRepository: NotificationTemplateRepository,
+    private environmentRepository: EnvironmentRepository
+  ) {}
+  async execute(command: ListWorkflowsCommand): Promise<ListWorkflowResponse> {
+    await this.validateEnvironment(command);
+    const res = await this.notificationTemplateRepository.getList(
+      command.user.organizationId,
+      command.user.environmentId,
+      command.offset,
+      command.limit,
+      command.searchQuery
+    );
+    if (res.data === null || res.data === undefined) {
+      return { workflowSummaries: [], totalResults: 0 };
+    }
+
+    return {
+      workflowSummaries: NotificationTemplateMapper.toWorkflowsMinifiedDtos(res.data),
+      totalResults: res.totalCount,
+    };
+  }
+
+  private async validateEnvironment(command: ListWorkflowsCommand) {
```

Comment:
Environment validation should be done at the middleware level during the JWT/API key resolution. It shouldn't be necessary at this stage.